### PR TITLE
Add UI for network mounts

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds UI for the APIs that where added in https://github.com/home-assistant/supervisor/pull/4269

<details>
<summary>Spoilers</summary>

No more will be provided.

</details>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by matthiasdebaat** [comment]
Great feature! One suggestion: can we move the 'move data disk' and 'add network link' buttons from the overflow menu to the bottom of the map? As we have for example on the hardware page. This makes it more findable.

File: src/panels/config/storage/ha-config-section-storage.ts [hunk 1]
```diff
@@ -100,14 +137,72 @@ class HaConfigSectionStorage extends LitElement {
...
+                                : nothing}${mount.type ===
+                              SupervisorMountType.NFS
+                                ? mount.path
+                                : mount.share}
+                            </span>
+                            <ha-svg-icon
+                              slot="meta"
+                              .mount=${mount}
+                              .path=${mdiReload}
+                              @click=${this._reloadMount}
```
Comment (bramkragten) [inline src/panels/config/storage/ha-config-section-storage.ts:185]: Should be `icon-button` then?

File: src/panels/config/storage/ha-config-section-storage.ts [hunk 2]
```diff
@@ -100,14 +137,72 @@ class HaConfigSectionStorage extends LitElement {
...
             : ""}
+          ${this._mounts?.length
+            ? html`
+                <ha-card
+                  outlined
+                  .header=${this.hass.localize(
+                    "ui.panel.config.storage.network_mounts.title"
+                  )}
+                >
+                  <div class="card-content">
```
Comment (bramkragten) [inline src/panels/config/storage/ha-config-section-storage.ts:148]:
```suggestion
```
This prevents double padding

**Review by bramkragten — DISMISSED** [review]

**Comment by piitaya** [comment]
Great!
During frontend meeting, we suggest "Network storage" instead of "Network mount" for the naming. What do you think?

**Comment by ludeeus** [comment]
Both are technical 🤷‍♂️

File: src/translations/en.json [hunk 3]
```diff
@@ -3969,6 +3973,59 @@
...
+              "media": "Media"
+            },
+            "mount_type": {
+              "nfs": "Network File Share (NFS)",
+              "cifs": "Samba/Windows (CIFS)"
+            },
+            "options": {
+              "name": {
+                "title": "Name",
+                "description": "This name will be shown to you in the  UI, and will also be the name of the folder created on your system"
```
Comment (frenck) [inline src/translations/en.json:3992]:
```suggestion
                "description": "This name will be shown to you in the UI, and will also be the name of the folder created on your system"
```

File: src/translations/en.json [hunk 4]
```diff
@@ -3969,6 +3973,59 @@
...
+                "title": "Remote share path",
+                "description": "This is the path of the remote share on your storrage server"
+              },
+              "type": {
+                "title": "Protocol",
+                "description": "This determine how you communicate with the storrage server"
+              },
+              "usage": {
+                "title": "Usage",
+                "description": "This determine how the share is intended to be used"
```
Comment (frenck) [inline src/translations/en.json:4012]:
```suggestion
                "description": "This determines how the share is intended to be used"
```

File: src/panels/config/storage/dialog-mount-view.ts [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+      {
+        name: "server",
+        required: true,
+        selector: { text: {} },
+      },
+      {
+        name: "type",
+        required: true,
+        type: "select",
+        options: [
```
Comment (frenck) [inline src/panels/config/storage/dialog-mount-view.ts:62]: Nit picking, maybe list CIFS first?

File: src/translations/en.json [hunk 6]
```diff
@@ -3969,6 +3973,59 @@
...
+                "title": "Server",
+                "description": "This is the domain name (FQDN) or IP address of the storage server you want to connect to"
+              },
+              "path": {
+                "title": "Remote share path",
+                "description": "This is the path of the remote share on your storage server"
+              },
+              "type": {
+                "title": "Protocol",
+                "description": "This determine how you communicate with the storage server"
```
Comment (frenck) [inline src/translations/en.json:4008]:
```suggestion
                "description": "This determines how to communicate with the storage server"
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**